### PR TITLE
More changes to GridEntry.

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Removes a number of other virtuals that weren't needed and dials in visibility.

Clarify terminology around notification and value ownership.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5682)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs [hunk 1]
```diff
@@ -1146,79 +1152,68 @@ public virtual IComponent[] GetComponents()
...
+        public object GetValueOwner() => _parent is null ? PropertyValue : _parent.GetValueOwnerInternal();
 
         /// <summary>
-        ///  Gets the owner of the current value.  This is usually the value of the
-        ///  root entry, which is the object being browsed.
+        ///  Gets the owner of the current value. This is usually the value of the root entry,
+        ///  which is the object being browsed.
         /// </summary>
-        public virtual object GetChildValueOwner(GridEntry childEntry) => PropertyValue;
+        /// <devdoc>
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs:1209]: not: old style tag

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyDescriptorGridEntry.cs [hunk 2]
```diff
@@ -528,83 +525,78 @@ public override void PaintLabel(Graphics g, Rectangle rect, Rectangle clipRect,
         {
             base.PaintLabel(g, rect, clipRect, selected, paintFullLabel);
 
-            IPropertyValueUIService propValSvc = PropertyValueUIService;
+            IPropertyValueUIService uiService = PropertyValueUIService;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyDescriptorGridEntry.cs:528]:
```suggestion
                        _propertyValueUIItems = PropertyValueUIService?.GetPropertyUIValueItems(this, _propertyInfo);
```

**Review by Tanya-Solyanik — APPROVED** [review]
looks good!

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs:1209]: Yeah, we can talk about this more in our next technical sync. There are comments that are specifically internal details that aren't part of the contract. I think this was the right tag for that.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyDescriptorGridEntry.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyDescriptorGridEntry.cs:528]: If we do that we'd be resetting the `_propertyValueUIItems` to null when we didn't before.

**Pull request merged**